src/main/java/ces/springmvc/bookmanagementspringmvc/controller/BookController.java

```diff
@@ -0,0 +1,139 @@
+package ces.springmvc.bookmanagementspringmvc.controller;
+
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import ces.springmvc.bookmanagementspringmvc.service.AuthorService;
+import ces.springmvc.bookmanagementspringmvc.service.BookService;
+import ces.springmvc.bookmanagementspringmvc.service.BookTypeService;
+import ces.springmvc.bookmanagementspringmvc.service.Book_BookTypeService;
+import ces.springmvc.bookmanagementspringmvc.validation.BookValidator;
+import ces.springmvc.bookmanagementspringmvc.constant.MVCCommandNames;
+import ces.springmvc.bookmanagementspringmvc.dto.AuthorDTO;
+import ces.springmvc.bookmanagementspringmvc.dto.BookDTO;
+import ces.springmvc.bookmanagementspringmvc.dto.BookTypeDTO;
+
+@Controller
+public class BookController {
+
+	@Autowired
+	private AuthorService authorService;
+
+	@Autowired
+	private BookService bookService;
+
+	@Autowired
+	private BookTypeService bookTypeService;
+
+	@Autowired
+	private Book_BookTypeService book_bookTypeService;
+
+	@GetMapping("/")
+	public String welcome(Model model) {
+
+		return "index";
+	}
+
+	@GetMapping("/books")
+	public String listBooks(Model model) {
+		List<BookDTO> listBook = bookService.getBooks();
+		model.addAttribute("listBook", listBook);
+
+		return MVCCommandNames.VIEW_BOOKS;
+	}
+
+	@GetMapping("/addBook")
+	public String showAddForm(Model model) {
+		model.addAttribute("url", MVCCommandNames.URL_SAVE_ADD);
+
+		_getFile(model);
+		return MVCCommandNames.VIEW_FORM;
+	}
+
+	@GetMapping("/editBook-id={bookId}")
+	public String showEditForm(Model model, @PathVariable(name = "bookId") Long bookId) {
+		try {
+			model.addAttribute("url", MVCCommandNames.URL_SAVE_EDIT);
+			model.addAttribute("bookEdit", bookService.getBookByBookId(bookId));
+			model.addAttribute("listBookTypeSelected", book_bookTypeService.getListBookTypeIdByBookId(bookId));
+
+			_getFile(model);
+
+			return MVCCommandNames.VIEW_FORM;
+
+		} catch (Exception e) {
+			logger.error("Sorry, something wrong!", e);
```

Comment:
for a more helpful log, you should add the error message in the log for more details, there are some other places that have the same issue.